### ISSUE TITLE
occasional missing header

### ISSUE THREAD

**Issue by leighlau** [brainstormforce/astra#2300]
Hi, should:
https://github.com/brainstormforce/astra/blob/1448c09babd0ae8e72c0fd31183f9ae12154f677/assets/js/unminified/frontend.js#L839
be: var window_width = window.innerWidth;

so the calculation is the same as used on line:
https://github.com/brainstormforce/astra/blob/1448c09babd0ae8e72c0fd31183f9ae12154f677/assets/js/unminified/frontend.js#L376

(the difference can cause the header to vanish for 'scroll bar width' amount of pixels)

or vice versa if you will

**Comment by luboslives** [comment]
To add, the mobile header remains on the page through 922px to 936px (but is hidden because of CSS) before it is swapped out by the desktop header in that `swap_header_on_breakpoint` function. So in that range we have no header shown on the page. Please make this simple change.

**Comment by IanLauwerys** [comment]
I am finding that at 939 width the desktop header is displayed as expected. They key piece of markup is:

```
<!-- #astra-fixed-header -->
<div id="ast-desktop-header">
```

At 938 width, the desktop header is swapped out for the mobile header thus:

```
<!-- #astra-fixed-header -->
<div id="ast-mobile-header" class="ast-mobile-header-wrap " data-type="off-canvas">
```

The mobile header is not shown though, due to the following CSS in frontend.min.css:

```
#ast-mobile-header {
    display: none;
}
```

At 920 width, the mobile header appears as the class ast-header-break-point is added to the <body> tag:
...
<body ... class="... ast-header-break-point ..." ...>
...

This enables the following CSS in frontend.min.css and the more specific selector overrides the display: none above making the mobile header visible:

```
.ast-header-break-point #ast-mobile-header {
    display: block;
}
```

I'm pretty sure the problem is failing to take in to account the scrollbar width when calculating breakpoint to apply  the ast-header-break-point class to <body>.

Pending a fix, a partial workaround is to add the following to Additional CSS in the customiser.

#ast-mobile-header{display:block;}

This ensures the mobile header is displayed as soon as it is swapped in at the breakpoint. It isn't a full solution as the mobile hamburger menu button is too close to the edge of the viewport until ast-header-break-point kicks in and other padding/margin elements that I haven't analysed are also applied. It also doesn't play well with certain menu settings when scrolling.

**Comment by rohitp-bsf** [comment]
Hello, 

Thank you for reporting this. This has been fixed and released in v3.3.0.

This issue was caused due to swapping of the header through JS and CSS which was causing the conflict in few cases as reported above.

For the solution, we have removed the entire JS code to swap the header and managing it through the CSS only with a media query.

**Comment by Nikschavan** [comment]
Closing the issue since this is fixed and released in v3.3.0
